pallets/compliance-manager/src/lib.rs

```diff
@@ -601,17 +598,15 @@ impl<T: Config> Module<T> {
         let complexity = asset_compliance
             .iter()
             .flat_map(|req| req.conditions())
-            .fold(0usize, |complexity, condition| {
-                let (claims, issuers) = condition.complexity();
-                complexity.saturating_add(claims.saturating_mul(match issuers {
-                    0 => default_issuer_count,
-                    _ => issuers,
-                }))
-            });
-        if let Ok(complexity_u32) = u32::try_from(complexity) {
-            if complexity_u32 <= T::MaxConditionComplexity::get() {
-                return Ok(());
-            }
+            .fold(0u32, |total, condition| {
+                let complexity = condition.complexity(default_issuer_count);
+                total.saturating_add(complexity)
+            })
+            // NB: If the compliance requirements are empty (0 complexity),
+            // then use the count of requirements.
+            .max(asset_compliance.len() as u32);
```

Comment:
Should this be a saturating add instead and if not, why?

Suppose we have 11 requirements. The first 10 are empty and return 0. The 11th is non-empty and returns 1 in complexity. Now all of a sudden the total complexity will be 1. Had the 11th been empty instead, the total complexity would be 11. That seems rather off to me. An edge-case to be sure, but still.

Reply:
This is to stop someone from adding a large number of empty requirements.  Users should not be adding empty compliance requirements.

Reply:
Alright but you could add a large number of empty requirements and only a single non-empty one and this logic wouldn't stop them, so it seems the logic doesn't cover the intent. We could keep count of how many empty there were and `saturating_add` those.

Reply:
If the total number of requirements is larger then the complexity of the non-empty requirements, then it will still use the "larger" value.  (i.e. 100 requirements and only one with complex 30, then the final complex would be 100).

Also note that the maximum complexity for asset compliance requirements is very low
(`T::MaxConditionComplexity == 50` right now).

Reply:
Originally I wanted to throw an error when the user tries to add empty compliance requirements, but that broke a lot of tests.  So to keep this PR simple, I added the above max logic.

I created a Jira ticket for using an error instead of this fix. MESH-1781

Reply:
Cool, good idea with the error long term 💯.
For now I think we can go with your code.